wit/core.wit

```diff
@@ -0,0 +1,89 @@
+package wasmcloud:core@0.1.0
+
+interface types {
+  type link-settings =          tuple<string, string>
```

Comment:
link settings is a `HashMap<string, string>` in rust, so this would have to be the (admittedly ugly) wit type of `list<tuple<string, string>>` which can get sugared into a map in Go and Rust provided the first type in the tuple is string

Reply:
yes....i have (since this commit) started a long detailed dive into this list<tuple wrap.  I have included the issue for tracking, but 100% agree